field_utils/FieldPointerUtilities.F90

```diff
@@ -151,6 +158,40 @@ subroutine assign_fptr_r8_rank2(x, fp_shape, fptr, rc)
       _RETURN(_SUCCESS)
    end subroutine assign_fptr_r8_rank2
 
+   subroutine assign_fptr_r4_rank3(x, fptr, rc)
+      type(ESMF_Field), intent(inout) :: x
+      real(kind=ESMF_KIND_R4), pointer, intent(out) :: fptr(:)
+      integer, optional, intent(out) :: rc
+
+      ! local declarations
+      type(c_ptr) :: cptr
+      integer(ESMF_KIND_I8), allocatable :: fp_shape(:)
+      integer :: status
+
+      fp_shape = get_fptr_shape(x, _RC)
+      call FieldGetCptr(x, cptr, _RC)
+      call c_f_pointer(cptr, fptr, fp_shape)
```

Comment:
This is a runtime failure unless `fp_shape` is size 1.  And we want size 3. (rank 3)

You need a test that covers this.

Reply:
Moved to `FieldCondensedArray.F90` and fixed with:
```
integer, parameter :: ARRAY_RANK = 3 ! FieldCondensedArray_private.F90
integer(ESMF_KIND_I8) :: fp_shape(ARRAY_RANK) ! FieldCondensedArray.F90
```
Now, it is not possible for fp_shape to not have the correct size.

Reply:
No.  That was not the intent.   The intent was to fix the procedure so that its argument was 3D.

Reply:
Note that the rank2 procedure that is still there has the same bug.